lib/SILOptimizer/Transforms/Devirtualizer.cpp

```diff
@@ -76,20 +75,10 @@ bool Devirtualizer::devirtualizeAppliesInFunction(SILFunction &F,
 
     Changed = true;
 
-    auto *AI = Apply.getInstruction();
-    if (!isa<TryApplyInst>(AI))
-      cast<SingleValueInstruction>(AI)->replaceAllUsesWith(NewInstPair.first);
-
-    DeadApplies.push_back(AI);
+    replaceDeadApply(Apply, NewInstPair.first);
```

Comment:
This is not code that I am that familiar with. But are you 200% sure that you are not going to run into any problems around iterator invalidation and or removing values still with uses. That is the reason why people usually follow this pattern.

Reply:
Other places in the code use replaceDeadApply(). Note that we don't have any active iterators in this loop. We're iterating over a vector of SILInstructions.

Reply:
I am not saying that using replaceDeadArray is necessarily wrong. What I am saying is that you are changing the code in a functional way by moving the time the applies are deleted from after the transformation to during the transformation. I am trying to prove to myself that doing so is safe (the pattern you are removing is a common defensive coding pattern in optimization passes). When I look at the git history these lines were added in the initial pass. @rudkx do you remember why you did this here?

Reply:
We'll see if the source compat suite passes :)

Reply:
That doesn't necessarily mean anything.